plugins/scaffolder-react/src/hooks/useEventStream.ts

```diff
@@ -169,67 +169,82 @@ export const useTaskEventStream = (taskId: string): TaskStream => {
-        subscription = observable.subscribe({
-          next: event => {
-            switch (event.type) {
-              case 'log':
-                return collectedLogEvents.push(event);
-              case 'cancelled':
-                dispatch({ type: 'CANCELLED' });
-                return undefined;
-              case 'completion':
-                emitLogs();
-                dispatch({ type: 'COMPLETED', data: event });
-                return undefined;
-              default:
-                throw new Error(
-                  `Unhandled event type ${event.type} in observer`,
-                );
+          // TODO(blam): Use a normal fetch to fetch the current log for the event stream
+          // and use that for an INIT_EVENTs dispatch event, and then
+          // use the last event ID to subscribe using after option to
+          // stream logs. Without this, if you have a lot of logs, it can look like the
+          // task is being rebuilt on load as it progresses through the steps at a slower
+          // rate whilst it builds the status from the event logs
+          const observable = scaffolderApi.streamLogs({ taskId });
+
+          const collectedLogEvents = new Array<LogEvent>();
+
+          function emitLogs() {
+            if (collectedLogEvents.length) {
+              const logs = collectedLogEvents.splice(
+                0,
+                collectedLogEvents.length,
+              );
+              dispatch({ type: 'LOGS', data: logs });
             }
-          },
-          error: error => {
-            emitLogs();
+          }
+
+          logPusher = setInterval(emitLogs, 500);
+
+          subscription = observable.subscribe({
+            next: event => {
+              switch (event.type) {
+                case 'log':
+                  return collectedLogEvents.push(event);
+                case 'cancelled':
+                  dispatch({ type: 'CANCELLED' });
+                  return undefined;
+                case 'completion':
+                  emitLogs();
+                  dispatch({ type: 'COMPLETED', data: event });
+                  return undefined;
+                default:
+                  throw new Error(
+                    `Unhandled event type ${event.type} in observer`,
+                  );
+              }
+            },
+            error: error => {
+              emitLogs();
+              // in some cases the error is a refused connection from backend
+              // this can happen from internet issues or proxy problems
+              // so we try to reconnect again after some time
+              // just to restart the fetch process
+              // details here https://github.com/backstage/backstage/issues/15002
+
+              if (!error.message) {
+                error.message =
+                  'We cannot connect at the moment, trying again in some seconds...';
+              }
+
+              setTimeout(() => {
+                startStreamLogProcess();
```

Comment:
I would feel better if we did not triggering indefinitely, perhaps try a couple of times and then give up?

Reply:
In my test case, this behavior does not happen because the following loop is connected or maybe the fetch error is raised (the connection isn't restored). But I see your worries, I will put a simple count to avoid this infinite retrying

Reply:
Done. I updated the message to show the counter too, so the user get a better feedback